src/envs/behavior_options.py

```diff
@@ -0,0 +1,1012 @@
+                            nonlocal plan_executed_forwards
+                            nonlocal tried_closing_gripper
+                            done_bit = False
+                            if (not plan_executed_forwards
+                                    and not tried_closing_gripper):
+                                # Step thru the plan to execute Grasping
+                                # phases 1 and 2
+                                ret_action = get_delta_low_level_hand_action(
+                                    env,
+                                    plan[0][0:3],
+                                    plan[0][3:6],
+                                    plan[1][0:3],
+                                    plan[1][3:6],
+                                )
+                                plan.pop(0)
+                                if len(plan) == 1:
+                                    plan_executed_forwards = True
+
+                            elif (plan_executed_forwards
+                                  and not tried_closing_gripper):
+                                # Close the gripper to see if you've gotten the
+                                # object
+                                ret_action = np.zeros(17, dtype=float)
+                                ret_action[16] = 1.0
+                                tried_closing_gripper = True
+
+                            else:
+                                # Grasping Phase 3: getting the hand back to
+                                # resting position near the robot.
+                                ret_action = get_delta_low_level_hand_action(
+                                    env,
+                                    reversed_plan[0][0:3],
+                                    reversed_plan[0][3:6],
+                                    reversed_plan[1][0:3],
+                                    reversed_plan[1][3:6],
+                                )
+                                reversed_plan.pop(0)
+                                if len(reversed_plan) == 1:
+                                    done_bit = True
+
+                            return ret_action, done_bit
+
+                        p.restoreState(state)
+                        p.removeState(state)
+
+                        return graspObjectOption
+
+                    print(f"PRIMITIVE: grasp {obj.name} fail, failed" +
+                          " to find plan to continuous params" +
+                          f" {grasp_offset}")
+                    return None
+
+                print(f"PRIMITIVE: grasp {obj.name} fail, too far")
+                return None
+            print(f"PRIMITIVE: grasp {obj.name} fail, too big or fixed")
+            return None
+        print(f"PRIMITIVE: grasp {obj.name} fail, no object")
+        return None
+    print(f"PRIMITIVE: grasp {obj.name} fail, agent already has an" +
+          " object in hand!")
+    return None
+
```

Comment:
So, in general the code pattern:
```
if not X:
    return None
if not Y:
    return None
if not Z:
    return None
W....
```

is preferable to:
```
if X:
    if Y:
        if Z:
            W
        return None
    return None
return None
```

because the latter is more readable and avoids code from getting over-indented like here. Could you try rewriting this function using this pattern instead?